src/providers/http_provider.py

```diff
@@ -42,13 +47,18 @@ def _get(self, url: str, params: Optional[dict] = None) -> Tuple[dict | list, di
                 timeout=self.REQUEST_TIMEOUT,
             )
 
+        if response.status_code != HTTPStatus.OK.value:
```

Comment:
Direct checking with enum also works
```suggestion
        if response.status_code != HTTPStatus.OK:
```